crypto/argon2/argon2.c

```diff
@@ -0,0 +1,195 @@
+/*
+ * Argon2 reference source code package - reference C implementations
+ *
+ * Copyright 2015
+ * Daniel Dinu, Dmitry Khovratovich, Jean-Philippe Aumasson, and Samuel Neves
+ *
+ * You may use this work under the terms of a Creative Commons CC0 1.0
+ * License/Waiver or the Apache Public License 2.0, at your option. The terms of
+ * these licenses can be found at:
+ *
+ * - CC0 1.0 Universal : http://creativecommons.org/publicdomain/zero/1.0
+ * - Apache 2.0        : http://www.apache.org/licenses/LICENSE-2.0
```

Comment:
When integrated into OpenSSL we will be using Apache 2.0, so I think this part of the comment should be removed.

Reply:
Will do.

Reply:
You shall do no such thing. That is the license of this file, no matter the integration. Removing this comment is tantamount to changing its license, which I would find unacceptable unless the copyright holders agree to it.

Reply:
What we *can* do, at least if changes are med within the project, is to *add* our own blurb at the top of the file.

Reply:
> You shall do no such thing. That is the license of this file, no matter the integration. Removing this comment is tantamount to changing its license, which I would find unacceptable unless the copyright holders agree to it.

I completely disagree @levitte. That comment says we get to choose whether to distribute it under Apache 2.0 or under CC0. That's exactly what we're doing. We're distributing it under Apache 2.0 (and no longer distributing it under CC0). Since we've made that choice the CC0 comment needs to be removed as it is no longer a correct description of the licensing state of this file. The copyright holders already did agree to this change by writing the comment that gave us that option in the first place!!!

Incidentally the CC0 license is effectively the same as putting this work in the public domain - the original copyright holders have waived all of their rights under copyright law. So we are at absolute liberty to make whatever changes we want to it - including to this comment.

Reply:
BTW - irrespective of all of the above, OpenSSL's CLA requirements mean that even if the work being integrated is in the public domain, we still need a CLA from the original copyright holders and their explicit consent to integrate it into OpenSSL. Therefore @levitte's requirement for "the copyright holders agree to it" will be met in any case.

It all still adds up to the same thing - we should remove this part of the comment.

Reply:
What jurisdiction are we talking about?  In most of Europe (save Commonwealth countries), there are rights that you cannot waive, by law.  CC0 doesn't change that.

Reply:
> What jurisdiction are we talking about? In most of Europe (save Commonwealth countries), there are rights that you cannot waive, by law. CC0 doesn't change that.

Well IANAL so I can't comment about whether CC0 *really* waives their rights or not. But it does not change then intent of the original authors that we can do whatever we like with this code.

Reply:
Does that include *removing a license*? Because there's a difference between the CC0 and public domain, in that the latter is not a license, it's more like a generic statement that there is no copyright claim on the material (side note: this works in commonwealth copyright law, but not with "droits d'auteur" style of law, which most of the rest of Europe implements)

Reply:
> Does that include removing a license? 

I believe so yes. If there is no copyright claim by the original authors then you are at liberty to relicense if you so wish.

Regardless of the CC0 argument though the authors give us explicit permission to pick one from Apache and CC0 (this is strictly speaking redundant because CC0 would enable you to do this anyway). We *are* picking one and not picking the other one. So our integrated work *must* remove the old option which we didn't pick. It is no longer a valid description of the licensing state.